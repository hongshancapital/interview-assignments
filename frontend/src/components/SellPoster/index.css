.sell-poster.dark {
  color: white;
}

.sell-poster.light {
  color: black;
}

.sell-poster li {
  list-style: none;
}

.sell-poster ul {
  padding: 0;
  margin: 0;
}

.sell-poster-content {
  height: 67%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-flow: column;
}

.sell-poster-title {
  font-size: 45px;
  font-weight: bold;
}

.sell-poster-des {
  font-size: 25px;
  margin-top: 20px;
}
