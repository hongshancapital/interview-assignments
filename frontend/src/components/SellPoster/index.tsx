import './index.css';

export interface SellPosterProps {
  title: string | string[];
  description?: string | string[];
  imgSrc: string;
  width?: string | number;
  height?: string | number;
  mode?: 'dark' | 'light';
  bgColor?: string;
}

const convertContent = (content: string | string[] | undefined) => {
  if (typeof content === 'string') {
    return [content];
  } else if (content === undefined) {
    return [];
  } else {
    return content;
  }
};

const SellPoster = ({
  title,
  description,
  imgSrc,
  width = '100%',
  height = '100%',
  mode = 'light',
  bgColor = '',
}: SellPosterProps) => {
  const titleArr = convertContent(title);
  const desArr = convertContent(description);

  return <div className={`sell-poster ${mode}`} style={{
    background: `${ bgColor } url('${imgSrc}') center center / auto 100% no-repeat`,
    width: typeof width === 'number' ? `${width}px` : width,
    height: typeof height === 'number' ? `${height}px` : height,
  }}>
    <div className='sell-poster-content'>
      <ul className="sell-poster-title">
        { titleArr.map((item, index) => <li key={index}>{item}</li>) }
      </ul>
      {!!desArr && <ul className="sell-poster-des">
        { desArr.map((item, index) => <li key={index}>{item}</li>) }
      </ul>}
    </div>
  </div>
};

export default SellPoster;