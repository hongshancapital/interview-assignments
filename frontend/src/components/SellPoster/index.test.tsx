import React from 'react';
import renderer from 'react-test-renderer';
import SellPoster from './index';
import iphoneSrc from '../../assets/iphone.png';

// 纯展示组件，测试时检查样式
test('SellPoster 样式检查', () => {
  const tree = renderer
    .create(<SellPoster width={1600} height={900} mode='dark' title={'xPhone'} description={['Lots to love. Less to spend.', 'Starting at $399.']} imgSrc={iphoneSrc}></SellPoster>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});