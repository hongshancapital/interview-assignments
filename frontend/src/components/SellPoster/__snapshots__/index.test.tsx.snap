// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellPoster 样式检查 1`] = `
<div
  className="sell-poster dark"
  style={
    Object {
      "background": " url('iphone.png') center center / auto 100% no-repeat",
      "height": "900px",
      "width": "1440px",
    }
  }
>
  <div
    className="sell-poster-content"
  >
    <ul
      className="sell-poster-title"
    >
      <li>
        xPhone
      </li>
    </ul>
    <ul
      className="sell-poster-des"
    >
      <li>
        Lots to love. Less to spend.
      </li>
      <li>
        Starting at $399.
      </li>
    </ul>
  </div>
</div>
`;
